### DISCUSSION TITLE
useConnect and useAccount loading is false on initial page load

### DISCUSSION THREAD

**Discussion by sammdec** [wevm/wagmi#37] [Ideas]
I kind of expected that on initial page load before the libraries have instantiated that the loading param in the hook would return true until it’s confirmed the user is connected or not.

**Comment by iduuck** [comment]
I think this should be an issue. But also, if not: What is the expected vs. actual behavior? I suppose you are not seeing the `loading` to `false` is it `undefined` or is it `true`?

---

@tmm I would suggest that we will be adding a `useInitialized` or something, so that the developer can implement a state for loading/initializing (e.g. the connect button).
I think this would be better, than having to rely on checking for something being `undefined`.

**Reply by sammdec** [reply]
The current behaviour is loading is false, I would expect it to be true while it’s initially loading.

**Reply by iduuck** [reply]
#I wouldn't be on the same page. My idea  for giving an opportunity for checking whether we are connected or not would be better. Setting loading to true wouldn't just not be the correct behavior nor would it be the correct value.

**Reply by sammdec** [reply]
It feels like when the page loads though that hook is loading to check if a user is connected. I would worry about adding complexity by adding more hooks just to check for initial page state

**Reply by tmm** [reply]
made a change that improves this behavior

https://github.com/tmm/wagmi/pull/79

**Reply by sudoFerraz** [reply]
I'm having the exact same issue but on useAccount instead of useConnect;
Seems like even if nothing has loaded yet, the useAccount hook loading var is still set to false on all cases.

**Comment by jem-computer** [comment]
I wonder if this is a similar category of issue - it looks like all of my hooks are running before I ever get a chainId - resulting in some funky behavior

here's how I have it patched for now
```ts
const provider = ({ chainId }: Config) => {
  if (chainId === undefined) {
    return null;
  }
  if (chainId === 1337) {
    return new providers.JsonRpcProvider();
  }

  return providers.getDefaultProvider(
    isChainSupported(chainId) ? chainId : chain.rinkeby.id,
    { infuraId, etherscanId }
  );
};
```
